NDTensors/ext/examples/NDTensorCUDA.jl

```diff
@@ -61,20 +62,17 @@ function main()
   #Currently this code fails with CUDA.allowscalar(false)
   # Because of outer calling the _gemm! function which calls a 
   # generic implementation
-  @allowscalar grad = gradient(f, cA, cB, cC, cD)
-  @allowscalar @test NDTensors.cpu(cB * cC * cD) ≈ NDTensors.cpu(grad[1])
+  grad = gradient(f, cA, cB, cC, cD)
+  @allowscalar @test cpu(cB * cC * cD) ≈ cpu(grad[1])
   @allowscalar @test (cB * cC * cD) ≈ grad[1]
   # Create a tuple of indices
-  decomp = (
-    dim(NDTensors.ind(grad[1], 1)),
-    dim(NDTensors.ind(grad[1], 2)) * dim(NDTensors.ind(grad[1], 3)),
-  )
+  decomp = (dim(ind(grad[1], 1)), dim(ind(grad[1], 2)) * dim(ind(grad[1], 3)))
   # Reshape the CuVector of data into a matrix
-  cuTensor_data = CUDA.reshape(NDTensors.data(storage(grad[1])), decomp)
+  cuTensor_data = CUDA.reshape(data(storage(grad[1])), decomp)
```

Comment:
```suggestion
  cuTensor_data = reshape(data(storage(grad[1])), decomp)
```
and same below.

Also, this code seems pretty sketchy, generally we should be calling `data(...)` in external code since that is an internal detail. Can we use `array` here or some other higher level function? We don't really want to encourage users to use `data(...)` by putting it in an example.

Reply:
Yeah I was worried about these lines especially with your changes to array tensor. I have modified these lines to use array and also remove the `storage` function.  Thanks!